WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt

```diff
@@ -219,13 +219,10 @@ class SiteCreationMainVM @Inject constructor(
 
     private fun clearOldSiteCreationState(wizardStep: SiteCreationStep) {
         when (wizardStep) {
-            SiteCreationStep.SITE_DESIGNS -> Unit // Do nothing
             SiteCreationStep.DOMAINS -> siteCreationState.domain?.let {
                 siteCreationState = siteCreationState.copy(domain = null)
             }
-            SiteCreationStep.SITE_PREVIEW -> Unit // Do nothing
-            SiteCreationStep.INTENTS -> Unit // Do nothing
-            SiteCreationStep.SITE_NAME -> Unit // Do nothing
+            else -> Unit // Do nothing
```

Comment:
This is just a suggestion and not a change request. Since the when clause has just two branches we could replace it with an `ifelse` clause. Alternatively we could group the full enumerated list eg.
```
        when (wizardStep) {
            SiteCreationStep.DOMAINS -> siteCreationState.domain?.let {
                siteCreationState = siteCreationState.copy(domain = null)
            }
            SiteCreationStep.SITE_PREVIEW, 
            SiteCreationStep.INTENTS,
            SiteCreationStep.SITE_NAME,
            SiteCreationStep.SITE_DESIGNS,
            SiteCreationStep.PROGRESS -> Unit // Do nothing
        }
```

Reply:
> This is just a suggestion and not a change request. Since the when clause has just two branches we could replace it with an ifelse clause. Alternatively we could group the full enumerated list eg.

Thanks! Replacing it with an `if` sounds good to me @antonis 👍🏻, that should be enough as we're not in a `return` statement so the `else` is actually optional. That way the code comment is embedded in the `if` condition so we can remove it as well, and if (ever) we have to handle more cases it's one IDE command away to turn it back into a `when` if that would make sense.

Reply:
Addressed in 1f7ef239